lib/active_merchant/billing/gateways/airwallex.rb

```diff
@@ -0,0 +1,249 @@
+        # These fields are required if billing data is sent.
+        card.first_name && card.last_name
+      end
+
+      def card_has_address_info(card, options)
+        # These fields are required if address data is sent.
+        options[:address1] && options[:country]
+      end
+
+      def add_address(card, options = {})
+        address = {}
+        address[:country_code] = options[:country]
+        address[:street] = options[:address1]
+        address[:city] = options[:city] if options[:city] # required per doc, not in practice
+        address[:postcode] = options[:zip] if options[:zip]
+        address[:state] = options[:state] if options[:state]
+        address
+      end
+
+      def add_invoice(post, money, options)
+        post[:amount] = amount(money)
+        post[:currency] = (options[:currency] || currency(money))
+      end
+
+      def add_card(post, card, options = {})
+        post['payment_method'] = {
+          'type' => 'card',
+          'card' => {
+            'expiry_month' => format(card.month, :two_digits),
+            'expiry_year' => card.year.to_s,
+            'number' => card.number.to_s,
+            'name' => card.name,
+            'cvc' => card.verification_value
+          }
+        }
+        add_billing(post, card, options)
+      end
+
+      def authorization_only?(options = {})
+        options.include?(:auto_capture) && options[:auto_capture] == false
+      end
+
+      def update_request_id(options, action)
+        options[:request_id] += "_#{action}"
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def commit(action, post, id = nil)
+        url = build_request_url(action, id)
+        post_headers = { 'Authorization' => "Bearer #{@access_token}", 'Content-Type' => 'application/json' }
+        response = parse(ssl_post(url, post_data(post), post_headers))
+
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: authorization_from(response),
+          avs_result: AVSResult.new(code: response['some_avs_response_key']),
+          cvv_result: CVVResult.new(response['some_cvv_response_key']),
+          test: test?,
+          error_code: error_code_from(response)
+        )
+      end
+
+      def handle_response(response)
+        case response.code.to_i
+        when 200...300, 400, 404
+          response.body
+        else
+          raise ResponseError.new(response)
+        end
+      end
+
+      def post_data(post)
+        post.to_json
+      end
+
+      def success_from(response)
+        %w(REQUIRES_PAYMENT_METHOD SUCCEEDED RECEIVED REQUIRES_CAPTURE CANCELLED).include?(response['status'])
+      end
+
+      def message_from(response)
+        response.dig('latest_payment_attempt', 'status') || response['status'] || response['message']
+      end
+
+      def authorization_from(response)
+        response.dig('latest_payment_attempt', 'payment_intent_id')
```

Comment:
Just bouncing around here, but if this fails to grab the `payment_intent_id` from the response, then the default value will just be `nil` in this case, and so subsequent transactions that rely on an `authorization` shouldn't yield a `blank` value.